### ISSUE TITLE
Diference between OpenStack's compute-hyperv repostory

### ISSUE THREAD

**Issue by johnwc** [cloudbase/compute-hyperv#6]
I noticed that this is not a fork of [OpenStacks compute-hyperv repository.](https://github.com/openstack/compute-hyperv) So what is the difference between these two? Which one is more active and updated?

**Comment by petrutlucian94** [comment]
This is a downstream version of the Hyper-V driver. It contains some backports of bug fixes that wouldn't be appropriate for the official driver (e.g. depend on some refactorings, require some config changes, etc).

**Comment by johnwc** [comment]
So, does the work on this driver slowly stream into the official driver? What would make a bug fix not appropriate for a official driver?

**Comment by petrutlucian94** [comment]
Most of the times, those are patches that merged upstream but for various reasons could not be backported to older branches.

For example, we had to avoid using some WMI providers (e.g. the failover cluster one, the iscsi initiator one, etc) as those were poorly implemented by MS (e.g. not propagating error codes, not retrieving all the requested data and other unexpected failures that did not occur after we switched to the underlying libraries used by those providers).

Now, switching APIs may be a bit troublesome as those aren't always identical (e.g. events provided by clusapi.dll callbacks don't include all the data that we get from WMI events), so there had to be some changes at the driver level. Such a fix can be difficult to backport upstream, requiring large refactorings in multiple projects (the hyper-v driver and os-win).

There were a few cases in which some features were requested by customers, landing here first and upstream afterwards. For example, some customers wanted to use one CSV per compute host, breaking some of the assumptions that nova made. They were using an older release, so we got the patch there first.
